### PR TITLE
Replace ALWAYS TRUE policy definitions with empty constraint expression

### PR DESCRIPTION
**Pull request by kamilczaja**
This PR changes the always true extension to create a policy definition with empty constraints instead of the artificial `ALWAYS_TRUE` constraint.
Included are also database migrations to change all existing always-true policies to the new format. Existing contract agreements are not affected.

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [ ] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have updated the Community Edition [Postman-Collection](https://github.com/sovity/edc-ce/blob/main/docs/api/postman_collection.json) if I changed existing APIs or added new APIs (e.g. for Management-API or API-Wrapper)
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: tests/src/test/java/de/sovity/edc/e2e/AlwaysTrueMigrationNewConsumerTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        val oldAlwaysTruePolicyOnProvider = providerClient.uiApi().getPolicyDefinitionPage().getPolicies().stream().filter(
+            policy -> policy.getPolicyDefinitionId().equals(AlwaysTruePolicyConstants.POLICY_DEFINITION_ID)
+        ).findFirst().orElseThrow();
+
+        val newAlwaysTruePolicyOnConsumer = consumerClient.uiApi().getPolicyDefinitionPage().getPolicies().stream().filter(
+            policy -> policy.getPolicyDefinitionId().equals(AlwaysTruePolicyConstants.POLICY_DEFINITION_ID)
+        ).findFirst().orElseThrow();
+
+        UiPolicyExpression oldAlwaysTrueExpression = oldAlwaysTruePolicyOnProvider.getPolicy().getExpression();
+        UiPolicyExpression newAlwaysTrueExpression = newAlwaysTruePolicyOnConsumer.getPolicy().getExpression();
```
Comment (ununhexium) [inline tests/src/test/java/de/sovity/edc/e2e/AlwaysTrueMigrationNewConsumerTest.java:45]: Could be combined with the 2 lines above.

File: tests/src/test/java/de/sovity/edc/e2e/AlwaysTrueMigrationNewProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class AlwaysTrueMigrationNewProviderTest {
+
+    @RegisterExtension
+    private static final E2eTestExtension E2E_TEST_EXTENSION = new E2eTestExtension(
+        "classpath:db/additional-test-data/alwaysTruePolicyMigrationTest",
+        ""
+    );
```
Comment (ununhexium) [inline tests/src/test/java/de/sovity/edc/e2e/AlwaysTrueMigrationNewProviderTest.java:26]: Builder here too.

File: tests/src/test/java/de/sovity/edc/e2e/AlwaysTrueMigrationNewProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        val newAlwaysTruePolicyOnProvider = providerClient.uiApi().getPolicyDefinitionPage().getPolicies().stream().filter(
+            policy -> policy.getPolicyDefinitionId().equals(AlwaysTruePolicyConstants.POLICY_DEFINITION_ID)
+        ).findFirst().orElseThrow();
+
+        val oldAlwaysTruePolicyOnConsumer = consumerClient.uiApi().getPolicyDefinitionPage().getPolicies().stream().filter(
+            policy -> policy.getPolicyDefinitionId().equals(AlwaysTruePolicyConstants.POLICY_DEFINITION_ID)
+        ).findFirst().orElseThrow();
+
+        UiPolicyExpression oldAlwaysTrueExpression = oldAlwaysTruePolicyOnConsumer.getPolicy().getExpression();
+        UiPolicyExpression newAlwaysTrueExpression = newAlwaysTruePolicyOnProvider.getPolicy().getExpression();
```
Comment (ununhexium) [inline tests/src/test/java/de/sovity/edc/e2e/AlwaysTrueMigrationNewProviderTest.java:45]: Combine with stream above.

File: tests/src/test/java/de/sovity/edc/e2e/common/AlwaysTruePolicyMigrationCommonTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        // act
+        val transfer = scenario.transferAndAwait(
+            InitiateTransferRequest.builder()
+                .contractAgreementId(negotiation.getContractAgreementId())
+                .dataSinkProperties(
+                    Map.of(
+                        EDC_NAMESPACE + "baseUrl", destinationUrl,
+                        EDC_NAMESPACE + "method", HttpMethod.POST,
+                        EDC_NAMESPACE + "type", "HttpData"
```
Comment (ununhexium) [inline tests/src/test/java/de/sovity/edc/e2e/common/AlwaysTruePolicyMigrationCommonTest.java:38]: We have these keys in `Prop.Edc`. DRY.

**Review by ununhexium — DISMISSED** [review]

**Review by richardtreier — APPROVED** [review]
LGTM

**Pull request merged**